coriolis_openstack_utils/security_groups.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2018 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+from oslo_log import log as logging
+
+
+LOG = logging.getLogger(__name__)
+
+
+def list_security_groups(openstack_client, tenant_id, secgroup_name=None):
+    if secgroup_name is None:
+        if int(openstack_client.connection_info["identity_api_version"]) == 2:
+            return openstack_client.neutron.list_security_groups(
+                tenant_id=tenant_id)['security_groups']
+        return openstack_client.neutron.list_security_groups(
+            project_id=tenant_id)['security_groups']
+    else:
+        if int(openstack_client.connection_info["identity_api_version"]) == 2:
+            return openstack_client.neutron.list_security_groups(
+                tenant_id=tenant_id, name=secgroup_name)['security_groups']
+        return openstack_client.neutron.list_security_groups(
+            project_id=tenant_id, name=secgroup_name)['security_groups']
+
+
+def create_security_group(openstack_client, tenant_id, body):
+    if int(openstack_client.connection_info["identity_api_version"]) == 2:
+        body['tenant_id'] = tenant_id
+    else:
+        body['project_id'] = tenant_id
+    post_body = {"security_group": body}
+    secgroup_id = openstack_client.neutron.create_security_group(
+        post_body)['security_group']['id']
+
+    return secgroup_id
+
+
+def create_destination_security_group_rules(
```

Comment:
I'd rename this to `get_destination_secgroup_rules_params` or something else (it currently sounds like you're actually issuing the API calls to create the secgroups here).
Alternatively, you could leave the function name as is and also pass in an `OpenStackClient()` instance and issue the create API call(s) here. (i.e. merge this with `add_rules_to_secgroup()`)